src/ucp/tag/eager_rcv.c

```diff
@@ -345,20 +320,10 @@ ucp_tag_offload_eager_first_handler(ucp_worker_h worker, void *data,
     ucp_request_t *req;
     ucs_status_t status;
 
-    /* First part of the fragmented message. Initialize matching queue and pass
-     * pointer to it back to UCT, so it will be provided with the rest of
-     * message fragments. Immediate data (indicating sync send) is passed with
-     * last fragment only, so ack will be sent upon receiving of the last
-     * fragment.
-     */
-    priv                             = ucp_tag_eager_offload_priv(
-                                           tl_flags, data, length,
-                                           ucp_offload_first_desc_t);
-    matchq                           = ucs_unaligned_ptr(&priv->matchq);
-    *(ucp_tag_frag_match_t**)context = matchq;
-    priv->super.super.tag            = stag;
-    priv->total_length               = length; /* total length is not final at
-                                                * this point */
+    priv                  = ucp_tag_eager_offload_priv(tl_flags, data, length,
+                                                       ucp_offload_first_desc_t);
+    priv->super.super.tag = stag;
+    priv->total_length    = length; /* total length is not final at this point */
```

Comment:
these fields are copied by ucp_recv_desc_init (because of + priv_len)
can we set them after ucp_recv_desc_init so it will not copy them?